### DISCUSSION TITLE
webcam errors

### DISCUSSION THREAD

**Discussion by pgg** [vladmandic/human#366] [Ideas]
hi vlad, would you consider adding a way to know why the webcam failed in your webcam util? more specifically if it failed because the user denied permission? the needed code in the ```catch``` clause for the ```getUserMedia``` call is: ```if (err instanceof DOMException && err.name === 'NotAllowedError')```.  I would like to notify users if they blocked the webcam (maybe by mistake). . .

**Comment by vladmandic** [comment]
Yes, that's fair, l'll add it

**Reply by pgg** [reply]
great, thanks

**Comment by pgg** [comment]
hi vlad, it's been a couple of weeks, just checking if you had not maybe forgotten about this?

**Comment by vladmandic** [comment]
its part of 3.0.7 release from 2 weeks ago. i was sure i updated this thread with info, but its not here :(
anyhow, `human.webcam.start` now returns `promise<string>` instead of `void` where string will include status (prefixed with `webcam error` if there is error)

**Reply by pgg** [reply]
cool it see it, thanks
